File: math/primenumbers/389-find-the-difference/find-the-difference.js
/**
 * @param {string} s
 * @param {string} t
 * @return {character}
 */
var findTheDifference = function(s, t) {
    let map={}


    for(let i=0;i<t.length;i++){
        if(map[t[i]]){
            map[t[i]]++
        }
        else{
            map[t[i]]=1
        }
    }

    console.log(map)
for(let i=0;i<s.length;i++){
        if(map[s[i]]){
            map[s[i]]--
        }
    }

    
    for(let m in map){
        if(map[m]!=0){
            return m
        }
    }
};

File: math/primenumbers/389-find-the-difference/find-the-difference.java
class Solution {
    public char findTheDifference(String s, String t) {

        int[] arr=new int[26];
        int m=t.length();

        for(int i=0;i<m;i++){
            arr[t.charAt(i)-'a']++;
        }

        for(int i=0;i<m-1;i++){
            arr[s.charAt(i)-'a']--;
        }

        for(int i=0;i<arr.length;i++){
            if(arr[i]==1){
                return (char)(97+i);
            }
        }

        return ' ';
       
    }
}